pymc/tests/logprob/test_joint_logprob.py

```diff
@@ -309,3 +317,279 @@ def test_multiple_rvs_to_same_value_raises():
+    # ndim >=1 & ndim<1
+    assert _get_scaling(45, (2, 3), 1).eval() == 22.5
+    assert _get_scaling(45, (2, 3), 0).eval() == 45
+
+    # list or tuple tests
+    # total_size contains other than Ellipsis, None and Int
+    with pytest.raises(TypeError, match="Unrecognized `total_size` type"):
+        _get_scaling([2, 4, 5, 9, 11.5], (2, 3), 2)
+    # check with Ellipsis
+    with pytest.raises(ValueError, match="Double Ellipsis in `total_size` is restricted"):
+        _get_scaling([1, 2, 5, Ellipsis, Ellipsis], (2, 3), 2)
+    with pytest.raises(
+        ValueError,
+        match="Length of `total_size` is too big, number of scalings is bigger that ndim",
+    ):
+        _get_scaling([1, 2, 5, Ellipsis], (2, 3), 2)
+
+    assert _get_scaling([Ellipsis], (2, 3), 2).eval() == 1
+
+    assert _get_scaling([4, 5, 9, Ellipsis, 32, 12], (2, 3, 2), 5).eval() == 960
+    assert _get_scaling([4, 5, 9, Ellipsis], (2, 3, 2), 5).eval() == 15
+    # total_size with no Ellipsis (end = [ ])
+    with pytest.raises(
+        ValueError,
+        match="Length of `total_size` is too big, number of scalings is bigger that ndim",
+    ):
+        _get_scaling([1, 2, 5], (2, 3), 2)
+
+    assert _get_scaling([], (2, 3), 2).eval() == 1
+    assert _get_scaling((), (2, 3), 2).eval() == 1
+    # total_size invalid type
+    with pytest.raises(
+        TypeError,
+        match="Unrecognized `total_size` type, expected int or list of ints, got {1, 2, 5}",
+    ):
+        _get_scaling({1, 2, 5}, (2, 3), 2)
+
+    # test with rvar from model graph
+    with pm.Model() as m2:
+        rv_var = pm.Uniform("a", 0.0, 1.0)
+    total_size = []
+    assert _get_scaling(total_size, shape=rv_var.shape, ndim=rv_var.ndim).eval() == 1.0
+
+
+def test_joint_logp_basic():
+    """Make sure we can compute a log-likelihood for a hierarchical model with transforms."""
+
+    with pm.Model() as m:
+        a = pm.Uniform("a", 0.0, 1.0)
+        c = pm.Normal("c")
+        b_l = c * a + 2.0
+        b = pm.Uniform("b", b_l, b_l + 1.0)
+
+    a_value_var = m.rvs_to_values[a]
+    assert m.rvs_to_transforms[a]
+
+    b_value_var = m.rvs_to_values[b]
+    assert m.rvs_to_transforms[b]
+
+    c_value_var = m.rvs_to_values[c]
+
+    (b_logp,) = joint_logp(
+        (b,),
+        rvs_to_values=m.rvs_to_values,
+        rvs_to_transforms=m.rvs_to_transforms,
+        rvs_to_total_sizes={},
+    )
+
+    # There shouldn't be any `RandomVariable`s in the resulting graph
+    assert_no_rvs(b_logp)
+
+    res_ancestors = list(walk_model((b_logp,)))
+    assert b_value_var in res_ancestors
+    assert c_value_var in res_ancestors
+    assert a_value_var in res_ancestors
+
+
+@pytest.mark.parametrize(
+    "indices, size",
+    [
+        (slice(0, 2), 5),
+        (np.r_[True, True, False, False, True], 5),
+        (np.r_[0, 1, 4], 5),
+        ((np.array([0, 1, 4]), np.array([0, 1, 4])), (5, 5)),
+    ],
+)
+def test_joint_logp_incsubtensor(indices, size):
```

Comment:
This may be redundant? If not should be in another file (corresponding to where the inc_subtensor logp is defined)

Reply:
I put this one in `tests/logprob/test_rewriting.py`. Is this correct?